pkg/api/v1/interaction.proto

```
@@ -2,8 +2,22 @@ syntax = "proto3";
 option go_package = "thoreinstein.com/rig/pkg/api/v1;apiv1";
 
+// UIService allows plugins to call back into the host to perform UI operations.
+service UIService {
+  // Prompt asks the user for a text input.
+  rpc Prompt(PromptRequest) returns (PromptResponse);
+  // Confirm asks the user for a yes/no confirmation.
+  rpc Confirm(ConfirmRequest) returns (ConfirmResponse);
+  // Select asks the user to choose from a list of options.
+  rpc Select(SelectRequest) returns (SelectResponse);
+  // UpdateProgress provides real-time status updates for a long-running task.
+  rpc UpdateProgress(ProgressUpdate) returns (google.protobuf.Empty);
```

Buf (1.65.0)

[error] RPC request type "ProgressUpdate" should be named "UpdateProgressRequest" or "UIServiceUpdateProgressRequest".

(RPC_REQUEST_STANDARD_NAME)